Source/WebCore/Modules/cookie-store/CookieStore.cpp

```diff
@@ -56,25 +65,243 @@ namespace WebCore {
+            if (cookie.domain.utf8().length() > maximumAttributeValueSize) {
+                ensureOnContextThread([completionHandler = WTFMove(completionHandler), maxSize = maximumAttributeValueSize]() mutable {
+                    completionHandler(Exception { TypeError, makeString("The size of the domain must not be greater than ", maxSize, " bytes") });
+                });
+                return;
+            }
+        }
+
+        cookie.path = WTFMove(options.path);
+        if (!cookie.path.isNull()) {
+            if (!cookie.path.startsWith('/')) {
+                ensureOnContextThread([completionHandler = WTFMove(completionHandler)]() mutable {
+                    completionHandler(Exception { TypeError, "The path must begin with a '/'"_s });
+                });
+                return;
+            }
+
+            if (!cookie.path.endsWith('/'))
+                cookie.path = cookie.path + '/';
+
+            // FIXME: <rdar://85515842> Obtain the encoded length without allocating and encoding.
+            if (cookie.path.utf8().length() > maximumAttributeValueSize) {
+                ensureOnContextThread([completionHandler = WTFMove(completionHandler), maxSize = maximumAttributeValueSize]() mutable {
+                    completionHandler(Exception { TypeError, makeString("The size of the path must not be greater than ", maxSize, " bytes") });
+                });
+                return;
+            }
+        }
+
+        if (options.expires)
+            cookie.expires = *options.expires;
+
+        switch (options.sameSite) {
+        case CookieSameSite::Strict:
+            cookie.sameSite = Cookie::SameSitePolicy::Strict;
+            break;
+        case CookieSameSite::Lax:
+            cookie.sameSite = Cookie::SameSitePolicy::Lax;
+            break;
+        case CookieSameSite::None:
+            cookie.sameSite = Cookie::SameSitePolicy::None;
+            break;
+        }
+
+        auto& cookieJar = page->cookieJar();
+        auto resultHandler = [this, protectedThis = WTFMove(protectedThis), completionHandler = WTFMove(completionHandler)] (bool setSuccessfully) mutable {
+            ensureOnContextThread([completionHandler = WTFMove(completionHandler), setSuccessfully]() mutable {
+                if (!setSuccessfully)
+                    completionHandler(Exception { TypeError });
+                else
+                    completionHandler({ });
+            });
+        };
+
+        cookieJar.setCookieAsync(document, document->url(), cookie, WTFMove(resultHandler));
+    };
+
+    ensureOnMainThread(WTFMove(setCookie));
+}
+
+Ref<CookieStore> CookieStore::create(ScriptExecutionContext* context)
+{
+    auto cookieStore = adoptRef(*new CookieStore(context));
+    cookieStore->suspendIfNeeded();
+    return cookieStore;
+}
+
+CookieStore::CookieStore(ScriptExecutionContext* context)
+    : ActiveDOMObject(context)
+    , m_mainThreadBridge(MainThreadBridge::create(*this, context->identifier()))
```

Comment:
Good point. What should we do if context is null? Is there some error we should throw since we will be unable to get an identifier and thus, unable to construct a mainThreadBridge?

Reply:
Just use a default constructed context identifier (which is an invalid identifier)